docs/aqua-book/language/modularity.md

```diff
@@ -0,0 +1,159 @@
+
+const CONST_NAME = "something"
+
+service ServiceName:
+  do_something()
+  
+data MyType:
+  result: i32  
+
+func fn() -> string:
+  <- CONST_NAME
+```
+
+To declare everything contained in the file, use `declares *`:
+
+```aqua
+-- This module declares `CONST_NAME`, `ServiceName`, `MyType` and `fn`
+aqua AquaFile declares *
+
+const CONST_NAME = "something"
+
+service ServiceName:
+  do_something()
+  
+data MyType:
+  result: i32  
+
+func fn() -> string:
+  <- CONST_NAME
+```
+
+Note that symbols declared with `declares` are not exported to the host language. To export symbols to the host language, use [`export`](#exporting-to-the-host-language-with-export).
+
+## Importing other modules
+
+Aqua modules can import other modules to use their [declarations](#specifying-what-is-declared-by-the-module-with-declares). There are two ways to import a module: with [`import`](#with-import) and [`use`](#with-use).
+
+### With `import`
+
+The main way to import a module is via `import`. Everything declared in the imported module comes into the current namespace directly.
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua declares `Op`
+import "@fluencelabs/aqua-lib/builtin.aqua"
+
+func foo():
+  Op.noop()
+```
+
+It is possible to cherry-pick and rename imports using `import ... from`:
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua declares `Op`
+import Op as Noop from "@fluencelabs/aqua-lib/builtin"
+
+func foo():
+  Noop.noop()
+```
+
+### With `use`
+
+The `use` expression makes it possible to import a module as a named scope. The name of the scope is taken from [`aqua` header](#giving-a-name-to-an-aqua-module-with-aqua) of the imported module. Everything declared in the imported module is available in the current namespace as a member of the scope.
+
+```aqua
+aqua AquaFile declares foo
+
+-- builtin.aqua's header is `aqua BuiltIn declares *`
+use "@fluencelabs/aqua-lib/builtin.aqua"
+
+func foo():
+  BuiltIn.Op.noop()
+```
+
+It is possible to rename the imported module with `use ... as ...` expression:
+
+```aqua
+use "@fluencelabs/aqua-lib/builtin" as Renamed
+
+func foo():
+  Renamed.Op.noop()
+```
+
+Creation of a scope with `use` makes it easier to avoid name clashes and to understand where the symbol comes from. Thus it is recommended to prefer `use` instead of `import` when possible.
+
+### Imports resolution
+
+To learn how compiler resolves the import path, see [JS Aqua API](./../aqua-js-api.md).
+
+:::info
+`.aqua` extension in `import` and `use` expressions can be omitted. So, `import "builtin.aqua"` does exactly the same as `import "builtin"`.
+:::
+
+## Exporting to the host language with `export`
+
+While it's useful to split the code into several functions into different files, it's not always a good idea to compile everything into the host language.
+
+Another problem is libraries distribution. If a developer wants to deliver an Aqua library, she often needs to provide it in the compiled form as well.
```

Comment:
```suggestion
Another problem is libraries distribution. If a developer wants to deliver an Aqua library, it often needs to provide it in the compiled form as well.
```

Reply:
I believe we can't use "it" or "we" as a reference to "developer"

Reply:
for sure we can't use `it` but we can use `we` if sentence is modified a bit

Reply:
```suggestion
Another problem is libraries distribution. If we want to deliver an Aqua library, we often need to provide it in the compiled form as well.
```

I am also not completely sure about that sentence in general. What do you mean by `provide`. Shouldn't user compile his own aqua whenever needed? I don't think developers should distribute already compiled aqua e.g. on npm. Or you mean something different when saying `provide`

Reply:
I would probably remove the sentence and just focus on the important part of that if you want to use your aqua function (whether with `fluence run` or with `js-client`) - you will need to use `export` keyword

Reply:
Yeah, I was in doubt about this sentence too. It was taken from previous version without much modification. I believe, it might be quite common for a library to be an aqua library and meaningful js package at the same time. Take `aqua-ipfs` for example.